### PR TITLE
Updating functions of dragon package in order to read outputs of L10 box including metal grids

### PR DESCRIPTION
**Pull request by EMventura**
This is part of the stuff to do for the first release

### CODE REVIEW

File: dragons/meraxes/io.py [hunk 1]
```diff
@@ -845,11 +850,18 @@ def read_grid(fname, snapshot, name, h=None, h_scaling={}):
...
+            try:
+            	grid_dim = fin["InputParams"].attrs["ReionGridDim"][0]
+            except KeyError:
+            	grid_dim = fin["InputParams"].attrs["TOCF_HII_dim"][0]
+            ds_name = "Snap{:03d}/Grids/{:s}".format(snapshot, name)
+        elif spec == 1:
+            try:
+            	grid_dim = fin["InputParams"].attrs["MetalGridDim"][0]
+            except KeyError:
+            	grid_dim = fin["InputParams"].attrs["MetalGridDim"][0]
```
Comment (smutch) [inline dragons/meraxes/io.py:863]: What does this do? Looks the same as the initial try block?

File: dragons/meraxes/io.py [hunk 2]
```diff
@@ -879,12 +891,17 @@ def read_grid(fname, snapshot, name, h=None, h_scaling={}):
     return grid
 
 
-def list_grids(fname, snapshot):
+def list_grids(spec, fname, snapshot):
```
Comment (smutch) [inline dragons/meraxes/io.py:894]: For all of the functions that now expected the `spec` argument. Can you add a check that will throw an exception if `spec not in (0, 1)`?

**Review by smutch — CHANGES_REQUESTED** [review]
Looks good. Can you make a couple of wee changes though? Sorry to be a pain!

File: dragons/meraxes/io.py (hunk 2 above)
Comment (EMventura) [reply dragons/meraxes/io.py:894]: Sure thing!

File: dragons/meraxes/io.py (hunk 1 above)
Comment (EMventura) [reply dragons/meraxes/io.py:863]: You are absolutely right! I am removing this try-except block

**Comment by EMventura** [comment]
Changes made! Lemme know if you want me to do other changes

**Review by smutch — APPROVED** [review]

**Pull request merged**